typhon/signal.py

```diff
@@ -178,3 +184,98 @@ def _add_row(self, read, name, write=None):
+    """
+    Panel of Signals for Device
+    """
+
+    def __init__(self, parent=None, init_channel=None):
+        super().__init__(parent=parent)
+        # Create a SignalPanel layout to be modified later
+        self.setLayout(SignalPanel())
+        # Add default Kind values
+        self._kinds = dict.fromkeys([kind.name for kind in Kind], True)
+
+    def _get_kind(self, kind):
+        return self._kinds[kind]
+
+    def _set_kind(self, value, kind):
+        # If we have a new value store it
+        if value != self._kinds[kind]:
+            # Store it internally
+            self._kinds[kind] = value
+            # Remodify the layout for the new Kind
+            self._set_layout()
+
+    # Kind Configuration pyqtProperty
+    showHints = Property(bool,
+                         partial(_get_kind, kind='hinted'),
+                         partial(_set_kind, kind='hinted'))
+    showNormal = Property(bool,
+                          partial(_get_kind, kind='normal'),
+                          partial(_set_kind, kind='normal'))
+    showConfig = Property(bool,
+                          partial(_get_kind, kind='config'),
+                          partial(_set_kind, kind='config'))
+    showOmitted = Property(bool,
+                           partial(_get_kind, kind='omitted'),
+                           partial(_set_kind, kind='omitted'))
+
+    @Property(str)
+    def channel(self):
+        """The channel address to use for this widget"""
+        if self._channel:
+            return str(self._channel)
+        return None
+
+    @channel.setter
+    def channel(self, value):
+        if self._channel != value:
+            # Remove old connection
+            for channel in self._channels:
+                channel.disconnect()
+            self._channels.clear()
+            # Load new channel
+            self._channel = str(value)
+            channel = HappiChannel(address=self._channel,
+                                   tx_slot=self._tx)
+            self._channels.append(channel)
+            # Connect the channel to the HappiPlugin
+            channel.connect()
+
+    def add_device(self, device):
+        """Add a device to the widget"""
+        # Only allow a single device
+        self.devices.clear()
+        # Add the new device
+        super().add_device(device)
+        # Configure the layout for the new device
+        self._set_layout()
+
+    def _set_layout(self):
+        """Set the layout based on the current device and kind"""
+        # We can't set a layout if we don't have any devices
+        if not self.devices:
+            return
+        # Clear our layout
+        self.layout().clear()
+        shown_kind = [kind for kind in Kind if self._kinds[kind.name]]
+        # Iterate through kinds
+        for kind in (Kind.hinted, Kind.normal, Kind.config, Kind.omitted):
+            if kind in shown_kind:
+                for (attr, signal) in grab_kind(self.devices[0], kind.name):
```

Comment:
Do we want signals grouped by kind or by attribute here?

Reply:
🤔  My thought was you always want the higher priority signals at the top

Reply:
Sort order option? ;)

Reply:
I think this is correct. No right or wrong answer here and not hard to support both